### PR TITLE
add more IT for train/predict and train&predict action

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
add more IT for train/predict and train&predict action
 
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ml-algorithms/build.gradle [hunk 1]
```diff
@@ -10,6 +10,7 @@ plugins {
 }
 
 repositories {
+    mavenLocal()
```
Comment (Zhangxunmt) [inline ml-algorithms/build.gradle:13]: Is there something required to pull from the local or just make the process faster with local?

File: plugin/src/test/java/org/opensearch/ml/action/prediction/PredictionITTests.java [hunk 2]
```diff
@@ -5,107 +5,84 @@
...
+import com.google.common.collect.ImmutableList;
 
-    public void testTestingData() throws ExecutionException, InterruptedException {
-        verifyGeneratedTestingData(TESTING_DATA);
-        waitModelAvailable(taskId);
-    }
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.SUITE, numDataNodes = 2)
+public class PredictionITTests extends MLCommonsIntegTestCase {
+    private String irisIndexName;
+    private String modelId;
```
Comment (Zhangxunmt) [inline plugin/src/test/java/org/opensearch/ml/action/prediction/PredictionITTests.java:34]: I think these IT test should be in lieu of the UTs? though they don't show up in the jacoco test reports.

**Review by Zhangxunmt — APPROVED** [review]
Let's leave a comments in the build.gradle jacocoExclusions saying that the exclusives in the list are tests through IT tests ?

File: ml-algorithms/build.gradle (hunk 1 above)
Comment (ylwu-amzn) [reply ml-algorithms/build.gradle:13]: Ignore this part, this is for debugging the tribuo bug https://github.com/oracle/tribuo/pull/224. Some tests will fail. Will ignore the flaky part first to pass CI.

File: plugin/src/test/java/org/opensearch/ml/action/prediction/PredictionITTests.java (hunk 2 above)
Comment (ylwu-amzn) [reply plugin/src/test/java/org/opensearch/ml/action/prediction/PredictionITTests.java:34]: Let's keep both UT and IT. Jacoco may not cover the REST API integration test.

**Comment by ylwu-amzn** [comment]
> 

I think we'd better not say that. We should add more test to meet the jacoco coverage bar. It will be more clear and easy to work with community. Otherwise, if community contributor says their code covered by REST IT tests, it's hard to verify.

**Review by Zhangxunmt — APPROVED** [review]
LGTM

**Pull request merged**